### PR TITLE
first version of Geopackage read support

### PR DESCRIPTION
**Pull request by stempler**

### CODE REVIEW

File: io/plugins/eu.esdihumboldt.hale.io.geopackage/src/eu/esdihumboldt/hale/io/geopackage/internal/TableInstanceBuilder.groovy [hunk 1]
```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2013 Data Harmonisation Panel
```
Comment (thorsten-reitz) [inline io/plugins/eu.esdihumboldt.hale.io.geopackage/src/eu/esdihumboldt/hale/io/geopackage/internal/TableInstanceBuilder.groovy]: Old header

File: io/plugins/eu.esdihumboldt.hale.io.geopackage.test/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -0,0 +1,17 @@
+Manifest-Version: 1.0
+Bundle-ManifestVersion: 2
+Bundle-Name: Geopackage Tests
+Bundle-SymbolicName: eu.esdihumboldt.hale.io.geopackage.test
+Bundle-Version: 4.0.0.qualifier
+Bundle-Vendor: wetransform GmbH
+Bundle-RequiredExecutionEnvironment: JavaSE-1.8
+Require-Bundle: eu.esdihumboldt.hale.io.geopackage;bundle-version="4.0.0",
+ mil.nga.geopackage;bundle-version="3.4.0";visibility:=reexport,
```
[the remark this answers is not in the document]
Comment (stempler) [reply io/plugins/eu.esdihumboldt.hale.io.geopackage.test/META-INF/MANIFEST.MF]: No, I had changed the access rules in the project. Later I needed to do adaptions to the dependencies anyway (merging dependencies that offer the same packages), that might also fix that problem w/o the need for the access rules.

File: io/plugins/eu.esdihumboldt.hale.io.geopackage/src/eu/esdihumboldt/hale/io/geopackage/GeopackageInstanceReader.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+			Map<TypeDefinition, InstanceCollection> collections = new HashMap<>();
+
+			// try to load each feature table
+			for (String table : gpkg.getFeatureTables()) {
+				QName defTypeName = new QName(GeopackageSchemaBuilder.DEFAULT_NAMESPACE, table);
+				TypeDefinition type = getSourceSchema().getType(defTypeName);
+				if (!type.getConstraint(MappingRelevantFlag.class).isEnabled()) {
+					type = null; // only mapping relevant types allowed
+				}
+				if (type == null) {
```
[the remark this answers is not in the document]
Comment (stempler) [reply io/plugins/eu.esdihumboldt.hale.io.geopackage/src/eu/esdihumboldt/hale/io/geopackage/GeopackageInstanceReader.java]: What is returned in line 80 can already be null (so good catch, I need to take that into account), and the condition in line 81 may be true as well.

In general: When the Geopackage file's schema has not been loaded (but none or another schema is used).

[the remark this answers is not in the document]
Comment (stempler) [reply io/plugins/eu.esdihumboldt.hale.io.geopackage/src/eu/esdihumboldt/hale/io/geopackage/GeopackageInstanceReader.java]: Added the missing null check in line 81

File: platform/hale-platform.target [hunk 4]
```diff
@@ -45,8 +45,8 @@
 <repository location="http://build-artifacts.wetransform.to/p2/equinox-test/build_3"/>
 </location>
 <location includeAllPlatforms="false" includeConfigurePhase="false" includeMode="slicer" includeSource="true" type="InstallableUnit">
-<unit id="eu.esdihumboldt.hale.platform.feature.group" version="3.5.0.i20181217"/>
-<repository location="https://gitlab.wetransform.to/hale/hale-build-support/raw/d557fabee76729ce83ed5ec3e8c3e0cc950d40f5/updatesites/platform"/>
+<unit id="eu.esdihumboldt.hale.platform.feature.group" version="3.4.0.i202001141313"/>
```
[the remark this answers is not in the document]
Comment (stempler) [reply platform/hale-platform.target]: @florianesser Maybe that was a change you did not push? See https://github.com/halestudio/hale-platform/blame/master/build.gradle#L41

[the remark this answers is not in the document]
Comment (florianesser) [reply platform/hale-platform.target]: @stempler You are right, I made the change only in my fork and never created a pull request. Thanks for the pointer.

**Pull request merged**